lib/cusparse/extra.jl

```diff
@@ -19,41 +19,37 @@ for (bname,gname,elty) in ((:cusparseScsrgeam2_bufferSizeExt, :cusparseScsrgeam2
             descrB = CuMatrixDescriptor('G', 'L', 'N', index)
             descrC = CuMatrixDescriptor('G', 'L', 'N', index)
 
-            rowPtrC = CuArray{Int32,1}(undef, m+1)
+            rowPtrC = CuVector{Int32}(undef, m+1)
+            local colValC, nzValC
 
             function bufferSize()
-                out = Ref{Csize_t}(1)
+                out = Ref{Csize_t}()
                 $bname(handle(), m, n,
                     alpha, descrA, nnz(A), nonzeros(A), A.rowPtr, A.colVal,
                     beta, descrB, nnz(B), nonzeros(B), B.rowPtr, B.colVal,
-                    descrC, CuArray{$elty,1}(undef, 0), rowPtrC, CuArray{Int32,1}(undef, 0),
+                    descrC, CuPtr{$elty}(CU_NULL), rowPtrC, CuPtr{Int32}(CU_NULL),
```

Comment:
Can't you use CU_NULL directly? IIRC ccall will do the conversion to a differently-typed pointer automatically.